lib/src/css_box_widget.dart

```diff
@@ -59,8 +59,24 @@ class CssBoxWidget extends StatelessWidget {
     final direction = _checkTextDirection(context, textDirection);
     final padding = style.padding?.resolve(direction);
 
+    Width? maxWidthCalculated;
+    if (style.maxWidth != null && style.width != null) {
+      if (style.maxWidth!.unit == Unit.percent &&
+          style.width!.unit == Unit.px) {
+        // If our max is a percentage, we want to look at the size available and not be bigger than that.
+        try {
+          double width =
+              MediaQuery.of(context).size.width * (style.maxWidth!.value / 100);
+          maxWidthCalculated = Width(width, style.width!.unit);
+        } catch (_) {}
+      } else if (style.width!.unit == style.maxWidth!.unit &&
+          style.width!.value > style.maxWidth!.value) {
+        maxWidthCalculated = Width(style.maxWidth!.value, style.maxWidth!.unit);
+      }
+    }
```

Comment:
Could be extracted to a method called `calculateMaxWidth` and then you'd have `final maxWidthCalculated = _calculateMaxWidth(style);`

this way you don't have to use a variable (finals should be used whenever possible)